.github/workflows/build_test_publish.yml

```diff
@@ -0,0 +1,45 @@
+name: Build, Test, & Publish
+
+on:
+  push:
+    branches: [ "main" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v3
+    - name: Install OS Dependencies
+      run: apt-get install curl gzip 
+    - name: Set up Python 3.11
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.11"
+    - name: Upgrade pip
+      run: python -m pip install --upgrade build
+    - name: Download Extended Observatory Code File 
+      run: |
+        curl https://minorplanetcenter.net/Extended_Files/obscodes_extended.json.gz -o mpc_obscodes/obscodes_extended.json.gz
```

Comment:
I feel like the fetching and unzipping could be done in a python module / script in this repository. I know you linked to a project to run github actions locally, but I think it would be more portable for development if there was a simple script someone could run locally to fetch the file. Another nice-to-have.